Square incremental

brijeshphilz opened the pull request:
## What
Incremental sync for Customer and Order objects based on Start date. 

## How
Modified the existing Square connector for synching data.
 1) Order is incrementally synced base on modified date. For more API information refer https://developer.squareup.com/reference/square/orders-api/search-orders 

 2) Customer is incrementally synched based on Created Date.  https://developer.squareup.com/reference/square/customers-api/list-customers

Each request have max number of records based on Square API limitations.

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

brijeshphilz committed:
Square Connector Incremental changes

brijeshphilz committed:
Square Connector Incremental changes

brijeshphilz committed:
Square Connector Incremental changes - bug fix

brijeshphilz committed:
Square Connector Incremental changes - feedback

brijeshphilz committed:
Merge pull request #2 from airbytehq/master

Update to latest master branch

etsybaev committed:
🐛Destination-elasticsearch: enforce ssl connection on cloud (#18341)

* [16992] Destination-elasticsearch: enforce ssl connection on cloud

benmoriceau committed:
Bmoric/extract destination api (#18502)

* Tmp

* Extract the Attempt API from the V1 API

* Add comments

* Move Connection API out of configuration API

* format

* format

* Rename to Controller

* Rename to Controller

* Add values to the factory

* Change the constructor to use hadler instead of objects needed by the handler

* Update with new tags.

* Fix PMD errors

* Add explicit path to the controller

* Extract destiantion API

girarda committed:
bump to v15.0 (#18467)

* bump to v15.0

* disable backwards compatible test for 0.2.69

* fix spec

* skip comprehensive

* bump

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
Persist geography updates (#18501)

* persist geography in connection update

* use ApiPojoConverters

* add test coverage for persisting geography update

* persist geography column for workspace

* use US instead of AUTO in mock data so that tests don't pass due to the database-level default column value

* use ApiPojoConverters instead of Enum.convertTo in workspace handler

* format

ryankfu committed:
Add SSH tunnel for MongoDB Destination (#18280)

* Add SSH tunnel for MongoDB Destination

* Fixed host/port resolution issues with Docker containers

* Formatting changes

* Bump version number and adds documentation for SSH tunnel

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

malikdiarra committed:
Improve performance of the operation that retrieve all connector for a given definition (#18499)

swatosh commented on airbyte-integrations/connectors/source-square/integration_tests/configured_catalog.json:
```diff
@@ -148,13 +148,13 @@
-      "cursor_field": ["id"],
-      "destination_sync_mode": "overwrite"
+      "sync_mode": "incremental",
+      "cursor_field": ["created_at"],
```
Should the "cursor_field" here match the one at source.py line 393?

brijeshphilz replied:
yes. missed the file while committing changes.

swatosh reviewed (commented):
Only one question

isaacharrisholt committed:
🎉 New Source: News API (#18301)

* Initial commit

* Add streams

* Simplify using options in request_parameters

* Address comments

* Add examples to query

* Add docs

* Change authenticator to ApiKeyAuthenticator

* add seed config

* remove icon

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

artem1205 committed:
🐛 Source Hubspot: acceptance fix (#18498)

Commit:
🎉 New Source: Workable (low-code SDK) (#18033)

* Generate initial Workable connector

* Implement jobs route

* Make acceptance tests pass

* Implement pagination

* Try to fix pagination (not yet working)

* Make pagination work

* Add created_after parameter

* Add Workable docs

* Move unused test

* Add candidates stream

* Fix acceptance tests

* Remove unused files

* Add stages stream

* Add recruiters stream

* Remove sample_state file

* Add Workable to docs/integrations/README.md

* Update changelog

* Change Sentry page size back

* Rename created_after_date to start_date

* Remove data type map

* Add start_date to sample_config.json

* Add streams documentation to yaml

* add title to params inspec

* format files

* fix: added source to source_definitions

Co-authored-by: Harshith Mullapudi <harshithmullapudi@gmail.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

alafanechere committed:
source-airtable: improve discovery logic (#18491)

cirdes committed:
🎉 Destination Meilisearch: migrating to python and fix integration  (#18036)

* Add meilisearch python destionation

* add destination

* first working version

* Writting tests

* writer more resilient

* Adding unit test

* correct integration test stream param

* update doc

* remove unused variables

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

Commit:
🐛 Source File: add retry logic for `Connection reset by peer - 104` error (#18428)

Commit:
🪟  🐛  Split Heading and Text components (#17992)

* split Text and Heading components

* replace Text with Heading where appropriate

artem1205 committed:
🐛 Source Facebook Marketing: increase timeout (#18536)

benmoriceau committed:
Bmoric/extract db migration api (#18459)

* Tmp

* Extract the Attempt API from the V1 API

* Add comments

* Move Connection API out of configuration API

* format

* format

* Rename to Controller

* Rename to Controller

* Add values to the factory

* Change the constructor to use hadler instead of objects needed by the handler

* Update with new tags.

* tmp

* Fix PMD errors

* Extract DB migrator

* Add something that I forgot

* restore destination factory initialization

* format

teallarson committed:
🪟 Apply default sync mode logic to new streams (#18451)

* brute force solution

* cleanup

* testing and comment

* review cleanup

benmoriceau committed:
Bmoric/extract destination definition api (#18515)

* Tmp

* Extract the Attempt API from the V1 API

* Add comments

* Move Connection API out of configuration API

* format

* format

* Rename to Controller

* Rename to Controller

* Add values to the factory

* Change the constructor to use hadler instead of objects needed by the handler

* Update with new tags.

* tmp

* Fix PMD errors

* Extract DB migrator

* Add something that I forgot

* extract destination definition api

* restore destination factory initialization

* format

* format

* explicit path

edmundito committed:
🪟 🎨 Add `PillSelect` UI component (#18376)

* Add PillButton UI component
Update Text component to support xs size

* Update PillButton prop names

* Remove outline and tint from PillButton until it's resolved with design

* Add PillButton example to popout

* PillButton -> PillSelect
Remove story from Popout
Create PillSelect component that manages select logic
Make PillButton part of PillSelect

* Support text overflow in pill button

* Show tooltip on pill select multiple items

* Use border-thin var in PillButton style

* Update gap in PillButton to use spacing var

benmoriceau committed:
extract destination definition specification api (#18521)

* Tmp

* Extract the Attempt API from the V1 API

* Add comments

* Move Connection API out of configuration API

* format

* format

* Rename to Controller

* Rename to Controller

* Add values to the factory

* Change the constructor to use hadler instead of objects needed by the handler

* Update with new tags.

* tmp

* Fix PMD errors

* Extract DB migrator

* Add something that I forgot

* extract destination definition api

* restore destination factory initialization

* extract destination definition specification api

* format

* format

* format

* explicit path

tkorenko committed:
Restore docs build (#18551)

+ Corrects two errors, unblocks docs build

Commit:
add authentication info to local development guide (#18509)

* add authentication info to local development guide

* rephrase

edmundito committed:
🪟 🐛 Fix UI issues in connection streams table (#18544)

* Fix PathPopout styling
* Ensure that text overflow still works
* Only show tooltip when there is more than 1
* Use caret instad of sort icon so that it's centered

* Fix option content for SyncSettignsDropdown so that it fits

* Fix @forward on StreamHeader scss

* Rollback tooltip updates

tkorenko committed:
Simplify the OSS documentation deploy system (#2670) (#18436)

+ Updates tools/bin/deploy_docusaurus to work on Github Runners
+ Adds 'Deploy docs.airbyte.com' Github workflow

tkorenko committed:
Simplify the OSS documentation deploy system (#2670) (#18448)

+ Updates Airbyte Docs: informs about automated Docs site
  build and deployment.

davydov-d committed:
Source Iterable: distinguish 401 from empty stream (#18537)

* #829 source iterable: distinguish 401 from empty stream

* #829 source iterable: upd changelog

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

tuliren committed:
Databricks destination: introduce personal staging location (#18557)

* Update s3 parquet writer

- Rename gcsFileLocation.
- Merge local variables.

* Implement staging location getter

* Add integration test and fix post request

* Add comment and format code

Commit:
Ensure that restart delay is set before any other activity is executed (#18547)

* Ensure that restart delay is set before any other activity is executed

* Add version check

* Formatting

dainiussa committed:
🎉 New Source: ActiveCampaign (#18335)

* Init source ActiveCampaign

* Added documentation files

* Added contacts and campaigns streams

* Added deals stream

* added segments stream

* Added lists stream

* Formatted schemas

* Added references to API docs

* added forms stream

* Added pagination

* Updated documentation

* Formatted schemas

* Removed unnecessary files

* Added changelog

* Updated base URL

* Added sample config

* add source seed

* run format

* generate uuid for activecampaign

* auto-bump connector version

Co-authored-by: Rytis Mickevičius <rytis.mickevicius1@gmail.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

alafanechere committed:
normalization: Fix failing mypy checks (#18534)

gosusnp committed:
Use Protocol Versions in Syncs (#18324)

* Pass protocol version into IntegrationLauncherConfig

* Use VersionedStreamStreamFactory in AirbyteSource/Destination

* Add AirbyteMessageBufferedWriter

* Use VersionedBufferedWriter

Commit:
🪟 🎉  Client-side validation of cron expressions <= 1 hr for cloud (#18044)

* add feature for cron expression format

* add new validation function for sub one hr cron

* add feature-dependent validation of cronExpression

* add test coverage for create & edit connection forms

* enable sub one hour cron expressions on cloud temporarily

girarda committed:
Lowcode connectors: Fix few issues with stream slicers (#18529)

* Fix a few bugs

* bump

edgao committed:
Protocol change: Define a set of well-known data types (#17486)

* add WellKnownTypes.yaml

* rename to snakecase + put in airbyte-protocol

* add examples

* more descriptoins

* descriptions, more restrictions, better regex

* update documentation

* explicitly call out BC support

isaacharrisholt committed:
🎉 New Source: Coin API (#18302)

* Initial commit

* Add streams

* Correct acceptance test config

* Make secret

* Update docsurl

* Remove catalog.json

* Add second stream

* Add default to spec

* Change to start_date and end_date

* Add links to CoinAPI docs in spec

* Change to start_date and end_date

* Add docs

* add coin api to source seed file

* remove icon

* generate source spec for coin api

Co-authored-by: Vincent Koc <koconder@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

isaacharrisholt committed:
🎉 New Source: TVMaze Schedule (#18333)

* Initial commit

* Add streams

* Remove file and update country code description

* Add docs

* format

* correct schema

* add tvmaze schedule to source spec seed

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>
Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

rytismi committed:
🎉 New Source: MailerLite (#18336)

* Init source MailerLite

* Added documentation

* Removed unnecessary files and a comment

* Added changelog

* Added sample config

* Updated base URL

* format

* add mailerlite to source def seed

* update docker image tag version

* auto-bump connector version

Co-authored-by: Dainius Salkauskas <dainiuxazz@gmail.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

xlahuerta committed:
🎉 New Source: Nasa APOD [python cdk] (#18394)

* NASA APOD connector first implementation

* Fix Acceptance and Unit tests

* Fix DecrecationWarning on NoAuth

* Update changelog with PR id

* `date` input removal in favor of `start_date` + Minor fixes

* Connector renaming to support different Nasa streams

* Update docs and source definitions with connector rename

* Support incremental updates

* Fix typo on abnormal_state.json, making acceptance tests to fail

* auto-bump connector version

Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

evantahler committed:
Add missing generated python files (#18580)

deanmalan committed:
🎉 New Source - Lokalise [low-code SDK] (#18522)

* Add keys stream

* Fix keys stream for now

* Start to add languages stream

* Add docs

* Start to add pagination

* Remove paginator again

* Revert "Remove paginator again"

This reverts commit 3f8daace174473be767ef78d0fa852be81925203.

* Pagination seems to work

* Clean up yaml

* Increase max page size

* Docs improvements

* Add comments stream

* Add contributors stream

* Fix contributors stream

* Add translations source

* Add new acceptance test format

* Add abnormal state

* Add PR number

* run format

* simplify selector

* add eof

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

isaacharrisholt committed:
🎉 New Source: Oura (#18224)

* Initial commit

* Add streams

* Final tidying

* Make secret

* Remove start_date and end_date parameters

* Add docs

* update cdk version

* run format + add source def for oura

* run config process manually

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>

elliottrabac committed:
🎉 New Source: Ashby (#18334)

* create boilerplate

* configure connector

* add ashby documentation

* update airbyte docs

* configure tests

* Revert "update airbyte docs"

This reverts commit 7dcf2b2dad1a77007b64aa8453aeb1c27e74ae58.

* update airbyte documentation

* update catalogs

* add start date for specific streams

* update config files

* add ashby to source def seed

* update doc

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

haithem-souala committed:
🎉 New Source - Sonar Cloud (#18475)

* init commit

* add docs

* add docs

* fix comments

* change string to array

henriblancke committed:
🎉 New source: SFTP Bulk [python cdk] (#17691)

* [INIT] setup source-ftp

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [ADD] add logic

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [ADD] unit, integration, acceptance tests

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] update docs

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] clean up crew

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] update host to localhost

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] rename from ftp to sftp bulk

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [INIT] setup source-ftp

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [ADD] add logic

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [ADD] unit, integration, acceptance tests

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] update docs

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] clean up crew

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] update host to localhost

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] rename from ftp to sftp bulk

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* update branch

* [FIX] integration tests

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [FIX] acceptance test fixture

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* [UPD] change ftp port

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* add uuid to image build

* add source def

* [FIX] generate ssh keys

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>

* auto-bump connector version

Signed-off-by: Henri Blancke <blanckehenri@gmail.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

haithem-souala committed:
🎉 New Source: Mailjet Mail API (#18332)

* init commit

* add streams

* add schemas

* add docs

* add docs

* fix comments

* restore vscode config

* restore vscode config

* clean

* lint

* lint

* lint

* lint

* fix comments

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

marcosmarxm committed:
Sonar Cloud: add to source spec and publish connector (#18584)

* solve conflict

* solve conflict spec.yaml

* add sonar cloud to seed + format

* add eof

* generate config source spec

marcosmarxm committed:
Source Mailjet Mail: publish connector and additional format (#18586)

* init commit

* add streams

* add schemas

* solve conflict

* add docs

* fix comments

* restore vscode config

* restore vscode config

* clean

* lint

* lint

* lint

* lint

* fix comments

* run format

* update acceptance test

* update schema

* update acceptance test

* auto-bump connector version

Co-authored-by: Haithem SOUALA <haithem.souala@woopit.fr>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

lmossman committed:
Implement Connector Builder testing panel (first iteration) (#18449)

* [SPIKE] add Builder page with editor

* add YamlEditor component

* export colors from CSS

* move template into dedicated file

* fix initial load from local storage

* move download button into separate component

* fix stylelint

* remove console log

* add todo

* make monaco background transparent and apply gradient to parent div background

* clarify comment

* remove unnecessary 180deg

* lock the builder UI behind a feature

* use rgb instead of hex to fix stylelint

* use _colors.scss and disable hex length stylelint rule

* disable rule in file

* generify ConnectorDocumentationLayout into TwoPanelLayout

* use TwoPanelLayout for BuilderPage

* add darker gradient color to _colors.scss and use in gradient

* move /builder to /connector-builder

* rename TwoPanelLayout to ResizablePanels

* put gradient on parent container instead

* make lightOverlay actually overlay instead of un-rendering children

* add testing panel component and stream selector

* fix rotated text

* fix more styling

* remove unnecessary comment

* add starting flex to right panel

* replace react-select dropdown with headless-ui listbox

* add test controls and fix button icon styling

* add Configure Test Input button and modal

* add connector builder client

* move connector builder state into dedicated context

* fix bold prop in Text component to not throw error

* pass parsed yaml->json into query

* store config in connector builder state context

* add result display

* fix styles of result display

* move config menu into testing panel

* capitalize stream names

* undo changes to Text component, since that is being addressed in a separate PR

* fix more merge issues and fix generated schema usage

* capitalize header

* remove console log

* rename data to streamRead

* fix stylelint and use variables

* use another variable for spacing

* use stream name in state data object

* move ConnectorBuilderStateProvider into ConnectorBuilderPage

* fix sizing of modal

* fix styling of active stream selector option

* Update airbyte-webapp/src/components/StreamTestingPanel/ResultDisplay.tsx

Co-authored-by: Tim Roes <tim@airbyte.io>

* Update airbyte-webapp/src/components/StreamTestingPanel/ResultDisplay.tsx

Co-authored-by: Tim Roes <tim@airbyte.io>

* hide result display panel if no data

* remove button specificity styling hacks, use size prop instead

* Update airbyte-webapp/src/components/StreamTestingPanel/StreamSelector.tsx

Co-authored-by: Tim Roes <tim@airbyte.io>

* change logs to errors

Co-authored-by: Tim Roes <tim@airbyte.io>

grubberr committed:
Source: Google Search Console: validate custom_reports.dimension (#18568)

Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

alafanechere committed:
SAT: make `expect_records` mandatory in `high` `test_strictness_level` (#18497)

itaseskii committed:
🎉 New destination: Yugabytedb (#18039)

* add yugabytedb destination connector

* add dest def + add changelog

* auto-bump connector version

Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

xpuska513 committed:
fix worker volumeMounts (#18604)

sarafonseca-123 committed:
🎉  New Source: RD Station Marketing [python cdk] (#18348)

* Add new source: RD Station Marketing

* Update catalog with all streams

* Update changelog with the rigth PR reference number

* Minor fixes to RD Station docs

* Change field replication_start_date to start_date

* Fix unit tests after changing replication_start_date name

* Fix typo

* Refact next_page_token logic

* Fix unit tests

* Remove print function

* Update airbyte-cdk version to 0.2

* Update airbyte-integrations/connectors/source-rd-station-marketing/source_rd_station_marketing/streams.py

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>

* Change from yield to yield from

* Apply format

* add rd station to source def

* update doc changelog

* auto-bump connector version

Co-authored-by: Marcos Marx <marcosmarxm@users.noreply.github.com>
Co-authored-by: marcosmarxm <marcosmarxm@gmail.com>
Co-authored-by: Octavia Squidington III <octavia-squidington-iii@users.noreply.github.com>

tkorenko committed:
Simplify the OSS documentation deploy system (#2670) (#18598)

+ Adds Slack notification about failed Docs builds
  (slack notifications fall into "#oss-master-build-failure" for now)
+ (while here) Unbreaks docs build

ikeadeoyin committed:
improved the digitalocean deployment guide (#18566)

dizel852 committed:
fixed failed build (#18610)

brijeshphilz committed:
Merge pull request #4 from airbytehq/master

update from master
